src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -84,6 +85,17 @@ public function __construct(ResourceClassResolverInterface $resourceClassResolve
         $this->graphQlSubscriptionManager = $graphQlSubscriptionManager;
         $this->graphQlMercureSubscriptionIriGenerator = $graphQlMercureSubscriptionIriGenerator;
         $this->reset();
+
+        $urlencode = ExpressionFunction::fromPhp('urlencode');
+        $this->expressionLanguage->addFunction($urlencode);
+
+        $this->expressionLanguage->addFunction(
+            new ExpressionFunction('iri', static function (string $apiResource): string {
```

Comment:
I suggest to add the type of URL to generate (absolute, relative...) as an extra optional argument. Defaulting to absolute looks good to me.